src/samples/http/server-https.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief HTTPS sample
+ *
+ * Basic HTTPS server. It uses a given certificate to crypto its contents.
+ *
+ * How to generate a certificate:
+ *
+ * openssl req -new > new.cert.csr
+ * openssl rsa -in privkey.pem -out new.cert.key
+ * openssl x509 -in new.cert.csr -out new.cert.cert -req -signkey new.cert.key -days 3652 -sha1
+ *
+ * To test it:
+ *
+ * run ./server-https -p 8080 -d "Hello HTTPS" -c new.cert.cert -k new,.cert.key
```

Comment:
I'd guess that browser presents a warning message, it could be an opportunity to elaborate on that as well.